clis/teliod/src/config.rs

```diff
@@ -53,21 +54,46 @@ fn reconnect_after_expiry_default() -> Percentage {
 }
 
 #[derive(Serialize, Deserialize, Debug, Default)]
-pub struct NodeState {
+pub struct DeviceIdentity {
     pub hw_identifier: String,
     pub private_key: SecretKey,
-    pub public_key: PublicKey,
-    pub machine_identifier: String,
+    pub machine_identifier: Option<String>,
 }
 
-#[derive(PartialEq, Eq, Serialize, Deserialize, Debug)]
+impl DeviceIdentity {
+    pub async fn new(identity_path: &PathBuf) -> DeviceIdentity {
+        info!("Fetching config");
+
+        if let Ok(file) = fs::File::open(identity_path) {
+            debug!("found existing config.");
+            if let Ok(c) = serde_json::from_reader(file) {
+                return c;
+            }
+        }
+
+        // Generate Secret/public key pair
+        debug!("Generating secret key!");
+        let private_key = SecretKey::gen();
+        let public_key = private_key.public();
+
+        // Generate hw_identifier
+        debug!("Generating hw identifier");
+        let hw_identifier = format!("{}.{}", public_key, "openWRT");
```

Comment:
This most certainly would also be used for QNAP and docker, so maybe just "teliod"?